be/src/vec/functions/function_timestamp.cpp

```diff
@@ -214,6 +234,90 @@ struct MakeDateImpl {
+        if constexpr (std::is_same_v<DateValueType, VecDateTimeValue>) {
+            return {std::make_shared<DataTypeDateTime>(), std::make_shared<DataTypeString>()};
+        } else {
+            return {std::make_shared<DataTypeDateTimeV2>(), std::make_shared<DataTypeString>()};
+        }
+    }
+
+    static DataTypePtr get_return_type_impl(const DataTypes& arguments) {
+        if constexpr (std::is_same_v<DateValueType, VecDateTimeValue>) {
+            return make_nullable(std::make_shared<DataTypeDateTime>());
+        } else {
+            return make_nullable(std::make_shared<DataTypeDateTimeV2>());
+        }
+    }
+
+    static Status execute(FunctionContext* context, Block& block, const ColumnNumbers& arguments,
+                          size_t result, size_t input_rows_count) {
+        DCHECK_EQ(arguments.size(), 2);
+        ColumnPtr argument_columns[2];
+        auto null_map = ColumnUInt8::create(input_rows_count, 0);
+        argument_columns[0] =
+                block.get_by_position(arguments[0]).column->convert_to_full_column_if_const();
+        argument_columns[1] =
+                block.get_by_position(arguments[1]).column->convert_to_full_column_if_const();
+        auto datetime_column = static_cast<const ColumnVector<ArgType>*>(argument_columns[0].get());
+        auto str_column = static_cast<const ColumnString*>(argument_columns[1].get());
+        auto& rdata = str_column->get_chars();
+        auto& roffsets = str_column->get_offsets();
+
+        ColumnPtr res = nullptr;
+
+        if constexpr (std::is_same_v<DateValueType, VecDateTimeValue>) {
+            res = ColumnVector<Int64>::create();
+            executeImpl(datetime_column->get_data(), rdata, roffsets,
+                        static_cast<ColumnVector<Int64>*>(res->assume_mutable().get())->get_data(),
+                        null_map->get_data(), input_rows_count);
+        } else {
+            res = ColumnVector<UInt64>::create();
+            executeImpl(datetime_column->get_data(), rdata, roffsets,
+                        static_cast<ColumnVector<UInt64>*>(res->assume_mutable().get())->get_data(),
+                        null_map->get_data(), input_rows_count);
+        }
+
+        block.get_by_position(result).column =
+                ColumnNullable::create(std::move(res), std::move(null_map));
+        return Status::OK();
+    }
+
+    static void executeImpl(const PaddedPODArray<ArgType>& ldata, const ColumnString::Chars& rdata,
+                            const ColumnString::Offsets& roffsets, PaddedPODArray<ArgType>& res,
+                            NullMap& null_map, size_t input_rows_count) {
+        res.resize(input_rows_count);
+        for (size_t i = 0; i < input_rows_count; ++i) {
+            auto& dt = (DateValueType&)ldata[i];
+            const char* str_data = reinterpret_cast<const char*>(&rdata[roffsets[i - 1]]);
+            if (std::strncmp("year", str_data, 4) == 0) {
```

Comment:
Could you move this heavy `if-else` out of the for-loop?

Reply:
input is not a const param, it's could contains year, day ,second.....in one column